### PR TITLE
Make min and max work with MPIBlockVector when some blocks have size 0

### PR DESCRIPTION
**Pull request by michaelbynum**
## Summary/Motivation:
This PR ensures the `min` and `max` methods on `MPIBlockVector` work when some blocks have arrays of size 0.

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/contrib/pynumero/sparse/tests/test_mpi_block_vector.py [hunk 1]
```diff
@@ -169,6 +169,28 @@ def test_min(self):
         self.assertEqual(self.v1.min(), 0.0)
         self.assertEqual(self.v2.min(), 0.0)
 
+    def test_min_max_with_size0_blocks(self):
```
Comment (jsiirola) [inline pyomo/contrib/pynumero/sparse/tests/test_mpi_block_vector.py:172]: Can we also add a test to exercise the exceptions for empty vectors?

Comment (mrmundt) [reply pyomo/contrib/pynumero/sparse/tests/test_mpi_block_vector.py:172]: Seconded but this looks good otherwise.

**Review by mrmundt — APPROVED** [review]

**Review by jsiirola — APPROVED** [review]
